File: Problem/Easy/Problem_4/Solution.cpp
// Write the program to calculate sum of first N natural numbers
#include <iostream>

// Recursive function
long long sumNaturalNumbers(int n = 0, long long sum = 0)
{
    if (n == 0)
        return sum;
    return sumNaturalNumbers(n - 1, sum + n);
}

// Main function
int main()
{
    int n = 6;
    std::cout << sumNaturalNumbers(n);
    return 0;
}

File: Problem/Easy/Problem_4/Solution.py
# The Python3 program using for calculate the sum of first N natural numbers with recursion

# Recursive function
def sumFirstNaturalNumbers(n : int, sum = 0) -> int:
    if n == 0:
        return sum
    return sumFirstNaturalNumbers(n - 1, sum + n)

# Main
if __name__ == "__main__":
    n = 6
    print("Sum of N first natural numbers is {}".format(sumFirstNaturalNumbers(n)))